### DISCUSSION TITLE
WinGet in Win11 & Win10 registry differences

### DISCUSSION THREAD

**Discussion by mcalexneill1** [microsoft/winget-cli#3791] [Configuration]
I have been using WinGet to deploy packages silently across a network on Windows 10 with pretty great success, I've been able to use a custom variable in the PS script that pulls from the registry the AppPath for WinGet, that way I never need to call it version specific in the script. I've been pulling this regentry from HKLM\SOFTWARE\Microsoft\Windows\Current Version\AppPaths\winget.exe. However I've noticed in my testing of W11 that this regentry is no longer found in HKLM but instead in HKCU meaning I can't pull the registry information for a custom variable. I've done a fair bit of to and fro on it and any "solution" I can find for this ends up being overly complex in the long run and would require considerable upkeep. My team were looking to migrate to using WinGet more frequently once we've upgraded the business to W11 however this odd difference between the installation on the two operating systems certainly throws a spanner in the works. My question then is it possible to change something in the install so that the regkey that is created is made in HKLM as opposed to HKCU? any thoughts or tips are greatly appreciated.

**Comment by JohnMcPMS** [comment]
Have you tried using our PowerShell module?  Obviously there are a lot of specifics that I'm not aware of in your situation, so it might not work for you.  It does support usage from `SYSTEM` though, and given what you are doing here I suspect you might be running as such.
https://www.powershellgallery.com/packages/Microsoft.WinGet.Client

I'm confident that the registry key in question is considered an internal implementation detail, and thus is not likely to have a setting that drives it.  There are other means to find the location in PowerShell if you need to follow that route:
```PowerShell
> $PackageBasePath = (Get-AppxPackage -AllUsers Microsoft.DesktopAppInstaller).InstallLocation
> $WingetPath = (Get-ChildItem -Recurse -Path $PackageBasePath -Filter winget.exe).FullName
```
That finds the file, but I feel it is likely that anything you do from here is not supported.

**Reply by mcalexneill1** [reply]
Hi John, 

Thanks for the response, I've been working with something similar to that by doing this - 

`$AppInstallerFilePath = Get-ChildItem -Path 'C:\Program Files\WindowsApps\' -Include *DesktopAppInstaller* -Filter *x64__* -Recurse`

Given that the actual .exe doesn't move I can dig into WindowsApps and filter to what would be correct folder in there and then pass that variable through in the WinGet calls. Giving me this as a full script - 

`$AppInstallerFilePath = Get-ChildItem -Path 'C:\Program Files\WindowsApps\' -Include *DesktopAppInstaller* -Filter *x64__* -Recurse`
`echo $AppInstallerFilePath`
`& $AppInstallerFilePath\winget.exe install...`

So far this has worked for me network deployment on Windows 11 devices.
